[quant][graphmode][fx] Add additional_object_mapping argument to convert

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #46355 [quant][eagermode] Add additional_fuser_method_mapping to config
* #46346 [quant][graphmode][fx] Add support for additional_{fusion/quant}_pattern
* #46345 [quant][graphmode][fx] Add support for additional_fuse_method_mapping
* #46344 [quant][graphmode][fx] Add additional_qat_module_mapping
* **#46338 [quant][graphmode][fx] Add additional_object_mapping argument to convert**

Summary:
Should we merge quantized module and quantized operator configurations?

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317435](https://our.internmc.facebook.com/intern/diff/D24317435)

jerryzh168 committed:
[quant][graphmode][fx] Add additional_object_mapping argument to convert

Summary:
Should we merge quantized module and quantized operator configurations?

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add additional_object_mapping argument to convert"


Summary:
Should we merge quantized module and quantized operator configurations?

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317435](https://our.internmc.facebook.com/intern/diff/D24317435)

[ghstack-poisoned]

vkuzo commented on torch/quantization/quantize_fx.py:
```diff
@@ -253,6 +253,18 @@ def convert_fx(graph_module, inplace=False, debug=False, convert_custom_config_d
+          # addtional object (module/operator) mappings that will overwrite the default
+          # module mappingn
+          "additional_object_mapping": {
+             "static": {
+                FloatModule: QuantizedModule,
+                float_op: quantized_op
+             },
+             "dynamic": {
+                FloatModule: DynamicallyQuantizedModule,
+                float_op: dynamically_quantized_op
+             },
```
how would custom mappings for QAT, and/or for models which have some parts quantized dynamically and statically fit?

jerryzh168 replied:
you mean additional qat module mappings? it's added in the next PR. it's kind of independent from these, it happens before we insert observer, e.g. we swap nn.Conv2d to nn.qat.Conv2d

"static" and "dynamic" here are used during convert, e.g. nn.qat.Conv2d to nn.quantized.Conv2d

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add additional_object_mapping argument to convert"


Summary:
Should we merge quantized module and quantized operator configurations?

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317435](https://our.internmc.facebook.com/intern/diff/D24317435)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add additional_object_mapping argument to convert"


Summary:
Should we merge quantized module and quantized operator configurations?

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317435](https://our.internmc.facebook.com/intern/diff/D24317435)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add additional_object_mapping argument to convert"


Summary:
Should we merge quantized module and quantized operator configurations?

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317435](https://our.internmc.facebook.com/intern/diff/D24317435)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add additional_object_mapping argument to convert"


Summary:
Should we merge quantized module and quantized operator configurations?

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317435](https://our.internmc.facebook.com/intern/diff/D24317435)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add additional_object_mapping argument to convert"


Summary:
Should we merge quantized module and quantized operator configurations?

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317435](https://our.internmc.facebook.com/intern/diff/D24317435)

[ghstack-poisoned]
